src/pal/src/locale/unicode.cpp

```diff
@@ -1632,3 +1641,68 @@ WideCharToMultiByte(
     PERF_EXIT(WideCharToMultiByte);
     return retval;
 }
+
+extern char g_szCoreCLRPath[MAX_PATH];
+
+/*++
+Function :
+
+PAL_BindResources - bind the resource domain to the path where the coreclr resides
+
+--*/
+VOID
+PALAPI
+PAL_BindResources(IN LPCSTR lpDomain)
+{
+#ifndef __APPLE__
+    char coreCLRDirectoryPath[MAX_PATH];
+
+    DWORD size = FILEGetDirectoryFromFullPathA(g_szCoreCLRPath, MAX_PATH, coreCLRDirectoryPath);
+    _ASSERTE(size <= MAX_PATH);
+    bindtextdomain(lpDomain, coreCLRDirectoryPath);
```

Comment:
The failure would mean that the FILEGetDirectoryFromFullPathA returned a string longer than the coreCLRDirectoryPath, which should never happen since the g_szCoreCLRPath has also MAX_PATH length. That's why I've just put in the assert to handle a possible case when someone would change the g_szCoreCLRPath length.